### ISSUE TITLE
Empty Vale Report on GitHub but not locally

### ISSUE THREAD

**Issue by fjp** [errata-ai/vale-action#75]
Hi, I am trying to use the errata-ai/vale-action@reviewdog action on GHE but the output is empty:

![image](https://user-images.githubusercontent.com/1286618/187442351-a36ad418-9e7e-4aec-b85f-30e633f899c5.png)

Running Vale locally with `vale --glob='*.{md,txt}' .` results in a bunch of expected suggestions/errors in the output report.

The workflow steps for Vale are:

```
name: lint
on:
  pull_request:
    branches:
      - main
  
jobs:
  vale:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout repository
        uses: actions/checkout@v3
        with:
          fetch-depth: 0

      - name: Vale
        uses: errata-ai/vale-action@reviewdog
        with:
          debug: true
          fail_on_error: true
          reporter: github-pr-check
          # added (default), diff_context, file, nofilter
          filter_mode: added
          vale_flags: --glob="*.{md,txt}"

        env:
          # Required, set by GitHub actions automatically:
          # https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret
          GITHUB_TOKEN: ${{secrets.GITHUB_TOKEN}}
```

The output of the build step is

```
Run errata-ai/vale-action@reviewdog
/usr/bin/docker run --name e28767c079dabc6d4c2c9503e976582b2fe0_d627cd --label 85e287 --workdir /github/workspace --rm -e GITHUB_TOKEN -e INPUT_DEBUG -e INPUT_FAIL_ON_ERROR -e INPUT_REPORTER -e INPUT_FILTER_MODE -e INPUT_VALE_FLAGS -e INPUT_VERSION -e INPUT_FILES -e INPUT_LEVEL -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_RUN_ATTEMPT -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_REF_NAME -e GITHUB_REF_PROTECTED -e GITHUB_REF_TYPE -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_ACTION_REPOSITORY -e GITHUB_ACTION_REF -e GITHUB_PATH -e GITHUB_ENV -e GITHUB_STEP_SUMMARY -e RUNNER_OS -e RUNNER_ARCH -e RUNNER_NAME -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/opt/github-action-runner/_work/_temp/_github_home":"/github/home" -v "/opt/github-action-runner/_work/_temp/_github_workflow":"/github/workflow" -v "/opt/github-action-runner/_work/_temp/_runner_file_commands":"/github/file_commands" -v "/opt/github-action-runner/_work/docs/docs":"/github/workspace" 85e287:67c079dabc6d4c2c9503e976582b2fe0  "latest" "all" "true" "github-pr-check" "true" "error" "added" "--glob=\"*.{md,txt}\""
Installing Vale version 'latest' ...
/bin/tar xz --overwrite -C /github/home -f /opt/github-action-runner/_work/_temp/cbb8a33e-206a-4dbd-83a2-04b7f3d39c9d
Installed version '2.20.[1](.../runs/180368?check_suite_focus=true#step:4:1)' into '/github/home/vale'.
Using Vale 2.20.1
/github/home/vale --glob="*.{md,txt}" sync

Downloading packages [0/6] █                                             0% | 0s
                                                                                
 SUCCESS  Downloaded package 'Google'

Downloading packages [1/6] ████████                                     [17](.../runs/180368?check_suite_focus=true#step:4:17)% | 2s
                                                                                
 SUCCESS  Downloaded package 'proselint'

Downloading packages [2/6] ███████████████                              33% | 3s
                                                                                
 SUCCESS  Downloaded package 'write-good'

Downloading packages [3/6] ██████████████████████                       50% | 4s
                                                                                
 SUCCESS  Downloaded package 'alex'

Downloading packages [4/6] █████████████████████████████                67% | 5s
                                                                                
 SUCCESS  Downloaded package 'Readability'

Downloading packages [5/6] ████████████████████████████████████         83% | 6s
                                                                                
 SUCCESS  Downloaded package 'Joblint'

Downloading packages [6/6] ███████████████████████████████████████████ 100% | 7s
Vale set-up comeplete; using '--output=/lib/rdjsonl.tmpl,--glob="*.{md,txt}",.'.
Running vale with reviewdog 🐶 ...
  /github/home/vale --output=/lib/rdjsonl.tmpl --glob="*.{md,txt}" .
  /bin/reviewdog -f=rdjsonl -name=vale -reporter=github-pr-check -fail-on-error=true -filter-mode=added -level=info
  2022/08/30 12:54:06 [vale] reported: .../runs/[18](.../runs/180368?check_suite_focus=true#step:4:18)0369 (conclusion=success)
```

Do you have any suggestion what might cause the empty reports or how to debug it?

To me it seems Vale is executed in a directory that contains no content files.

Is there a way to check which path is used to execute vale or to pass the [`-tee` debugging option to reviewdog](https://github.com/reviewdog/reviewdog#debugging) to debug the issue?

**Comment by fjp** [comment]
An example of an empty report is found here: https://github.com/ros-mobile-robots/ros-mobile-robots.github.io/pull/5/checks?check_run_id=8102264712

Is there any setting I am missing?

**Comment by lithammer** [comment]
I had similar issues. And it seems to be related to how `vale_flags` is used. I looked into the implementation and here's what I think is going on. The flags string is split into an array based on whitespace:

https://github.com/errata-ai/vale-action/blob/8dd360d78a68bffbc265fb9ec7c1b0b36779b3e1/src/input.ts#L16

And the "splatted" into an argument list here:

https://github.com/errata-ai/vale-action/blob/8dd360d78a68bffbc265fb9ec7c1b0b36779b3e1/src/input.ts#L78-L81

So what you end up with in the end is something like this:

```js
["--output=/lib/rdjsonl.tmpl", "--glob=\"*.{md,txt}\""]
```

The problem with this is the glob pattern will actually include the quotes. And that obviously doesn't match anything.

The solution is either of these two (at least for me):

```diff
-vale_flags: --glob="*.{md,txt}"
+vale_flags: --glob=*.{md,txt}
```

```diff
-vale_flags: --glob="*.{md,txt}"
+vale_flags: --glob *.{md,txt}
```

**Comment by unennhexium** [comment]
Same issue. Any suggestions?

<details><summary><b>Workflow file</b></summary>
    
```yml
name: Vale check

on:
  push:

jobs:
  vale:

    name:    vale
    runs-on: ubuntu-latest
    steps:

      - name: Checkout code
        uses: actions/checkout@v4.2.2

      - name: Check style
        uses: errata-ai/vale-action@v2.1.1
        with:

          files: 'sort/src,bench/benches,service/src,python/source,README.md'
          separator:  ','
          reporter:   github-check
          filter_mode: nofilter
          vale_flags: '--minAlertLevel=suggestion --glob=*.{rs,py,md}'
```

</details> 

### CI run

```
Running vale...
  /home/runner/vale --output=/home/runner/work/_actions/errata-ai/vale-action/d89dee975228ae261d22c15adcd03578634d429c/lib/rdjsonl.tmpl --minAlertLevel=suggestion --glob=*.{rs,py,md} sort/src bench/benches service/src python/source README.md
  Calling reviewdog 🐶
  /home/runner/reviewdog -f=rdjsonl -name=vale -reporter=github-check -fail-on-error=false -filter-mode=nofilter -level=info
  reviewdog: Reporting results for "vale"
  reviewdog: No results found for "vale". 0 results found outside diff.
```

### Local run

<details><summary><b>Output</b></summary>

```
$ vale --minAlertLevel=suggestion sort/src bench/benches service/src python/source README.md

 sort/src/lib.rs
 21:42  warning  Consider using 'primary',       alex.Race            
                 'hub', or 'reference' instead                        
                 of 'master'.                                         
 26:68  warning  In general, don't use an        Google.Ellipses      
                 ellipsis.                                            
 26:68  error    Consider using the '…' symbol   proselint.Typography 
                 instead of '...'.                                    
 27:1   error    Did you really mean 'Supress'?  Vale.Spelling        


 sort/src/library/ch_07.rs
 273:11  warning     Don't put a period at the end   Google.HeadingPunctuation 
                     of a heading.                                             
 461:17  warning     Don't put a period at the end   Google.HeadingPunctuation 
                     of a heading.                                             
 554:63  suggestion  Use semicolons judiciously.     Google.Semicolons         
 555:39  suggestion  Use semicolons judiciously.     Google.Semicolons         
 557:22  suggestion  Use semicolons judiciously.     Google.Semicolons         
 558:28  suggestion  Use semicolons judiciously.     Google.Semicolons         
 564:11  warning     Don't put a period at the end   Google.HeadingPunctuation 
                     of a heading.                                             


 sort/src/library/ch_08_04.rs
 41:7    error       Did you really mean 'a_i'?      Vale.Spelling     
 160:33  suggestion  Use semicolons judiciously.     Google.Semicolons 
 207:92  suggestion  Use semicolons judiciously.     Google.Semicolons 
 256:46  error       Commas and periods go inside    Google.Quotes     
                     quotation marks.                                  
 256:64  suggestion  Use semicolons judiciously.     Google.Semicolons 

 sort/src/library/mod.rs
 40:105  suggestion  Use semicolons judiciously.  Google.Semicolons 
 41:31   suggestion  Use semicolons judiciously.  Google.Semicolons 
 43:64   suggestion  Use semicolons judiciously.  Google.Semicolons 


 sort/src/library/ch_08_02.rs
 184:47  warning     'it is' is too wordy.           write-good.TooWordy 
 188:11  warning     Avoid first-person pronouns     Google.FirstPerson  
 574:17  suggestion  Use semicolons judiciously.     Google.Semicolons   
 575:83  suggestion  Use semicolons judiciously.     Google.Semicolons   
 576:18  suggestion  Use semicolons judiciously.     Google.Semicolons   


 sort/src/library/ch_08_90.rs
 13:11   warning     Don't put a period at the end   Google.HeadingPunctuation 
                     of a heading.                                             
 121:11  warning     Don't put a period at the end   Google.HeadingPunctuation 
                     of a heading.                                             
 284:11  warning     Don't put a period at the end   Google.HeadingPunctuation 
                     of a heading.                                             
 352:25  warning     Be careful with 'jugs', it's    alex.ProfanityUnlikely    
                     profane in some cases.                                    
 424:20  suggestion  Use semicolons judiciously.     Google.Semicolons         
 425:33  suggestion  Use semicolons judiciously.     Google.Semicolons         
 426:14  error       Did you really mean             Vale.Spelling             
                     'Uncomment'?                                              
 426:30  warning     Use 'preceding' instead of      Google.WordList           
                     'above'.                                                  
 446:36  warning     'enumerate' is too wordy.       write-good.TooWordy       
 447:71  suggestion  Use semicolons judiciously.     Google.Semicolons         
 462:40  suggestion  Use semicolons judiciously.     Google.Semicolons         
 480:75  suggestion  Use semicolons judiciously.     Google.Semicolons         
 488:52  suggestion  Use semicolons judiciously.     Google.Semicolons         
 489:46  suggestion  Use semicolons judiciously.     Google.Semicolons         
 490:76  suggestion  Use semicolons judiciously.     Google.Semicolons         
 504:47  suggestion  Use semicolons judiciously.     Google.Semicolons         
 507:18  error       Commas and periods go inside    Google.Quotes             
                     quotation marks.                                          
 510:18  error       Commas and periods go inside    Google.Quotes             
                     quotation marks.                                          
 510:57  suggestion  Use semicolons judiciously.     Google.Semicolons         
 581:62  warning     'be skipped' may be passive     write-good.Passive        
                     voice. Use active voice if you                            
                     can.                                                      
 581:62  suggestion  In general, use active voice    Google.Passive            
                     instead of passive voice ('be                             
                     skipped').                                                
 593:10  error       Consider using the '×' symbol   proselint.Typography      
                     instead of '0x0'.                                         

 bench/benches/wall_clock_criterion.rs
 13:12  error       Did you really mean 'pcd'?      Vale.Spelling          
 14:50  suggestion  Use semicolons judiciously.     Google.Semicolons      
 15:30  suggestion  Use semicolons judiciously.     Google.Semicolons      
 22:33  suggestion  Use semicolons judiciously.     Google.Semicolons      
 28:51  suggestion  Use semicolons judiciously.     Google.Semicolons      
 39:30  suggestion  Use semicolons judiciously.     Google.Semicolons      
 77:22  warning     'multiple' is too wordy.        write-good.TooWordy    
 85:25  warning     'equivalent' is too wordy.      write-good.TooWordy    
 86:68  warning     'equivalent' is too wordy.      write-good.TooWordy    
 87:11  warning     Be careful with 'execution',    alex.ProfanityUnlikely 
                    it's profane in some cases.                            

✖ 36 errors, 19 warnings and 31 suggestions in 19 files.
```

</details>
